ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.hpp

```diff
@@ -49,6 +49,18 @@ operation::ProgramWithCallbacks reduce_scatter_with_workers(
 }
 }; // namespace ccl
 
+ReduceScatter create_reduce_scatter_struct (
+    const Tensor& input_tensor,
+    const ttnn::operations::binary::BinaryOpType binary_op_type,
+    const uint32_t scatter_dim,
+    const uint32_t num_links,
+    const MemoryConfig output_mem_config,
+    const std::optional<size_t> user_defined_num_workers,
+    const std::optional<size_t> user_defined_num_buffers_per_channel,
+    const std::vector<Device*>& devices,
+    const ttnn::ccl::Topology topology
+);
+
```

Comment:
Would this be better to be inside `ccl::reduce_scatter_detail` namespace?

Reply:
yes I think that is a good recommendation.